packages/desktop-client/src/components/ColumnWidthContext.jsx

```diff
@@ -0,0 +1,186 @@
+import React, {
+  createContext,
+  useContext,
+  useState,
+  useCallback,
+  useEffect,
+} from 'react';
+
+import { useLocalPref } from '../hooks/useLocalPref';
+
+const ColumnWidthContext = createContext();
+
+export const ColumnWidthProvider = ({ children, prefName }) => {
+  const [columnSizePrefs, setColumnSizePrefs] = useLocalPref(prefName);
+  const [columnWidths, setColumnWidths] = useState({});
+  const [fixedSizedColumns, setFixedSizedColumns] = useState({});
+  const [, setPositionAccumulator] = useState(0);
+  const [clientWidth, setClientWidth] = useState(0);
+
+  useEffect(() => {
+    if (columnSizePrefs) {
+      setColumnWidths(JSON.parse(columnSizePrefs));
+    }
+  }, [columnSizePrefs]);
+
+  const totalWidth = useCallback(() => {
+    const currentTotalWidth = Object.values(columnWidths).reduce(
+      (acc, width) => acc + width,
+      0,
+    );
+
+    const otherColumnsWidth = Object.entries(fixedSizedColumns)
+      .filter(([key]) => !(key in columnWidths))
+      .reduce((acc, [, width]) => acc + width + 10, 0);
+
+    const widthSum = otherColumnsWidth + currentTotalWidth + 20;
+    return widthSum > clientWidth ? widthSum : clientWidth;
+  }, [columnWidths, fixedSizedColumns, clientWidth]);
+
+  const savePrefs = useCallback(() => {
+    setColumnSizePrefs(JSON.stringify(columnWidths));
+  }, [columnWidths, setColumnSizePrefs]);
+
+  const removeColumn = columnName => {
+    const { [columnName]: _, ...newObj } = columnWidths;
+    setColumnWidths(newObj);
+    savePrefs();
+  };
+
+  const handleDoubleClick = columnName => {
+    updateColumnWidth(columnName, -1);
+
+    setTimeout(() => {
+      let maximum = -1;
+      document
+        .querySelectorAll(`[data-resizeable-column=${columnName}]`)
+        .forEach(row => {
+          const rect = row.getBoundingClientRect();
+          const styles = getComputedStyle(row);
+          const localValue = Math.max(
+            rect.width +
+              parseFloat(styles.marginLeft) +
+              parseFloat(styles.marginRight),
+            110,
+          );
+
+          maximum = Math.max(localValue, maximum);
+        });
+
+      updateColumnWidth(columnName, maximum);
+
+      savePrefs();
+    }, 200);
+  };
```

Comment:
this doesn't work because when a column size is set to -1
```
updateColumnWidth(columnName, -1);
```
it sets the size of the div as "width: auto" after a rerender. this way we can measure biggest and set as the new column value. that's why it waits 100 ms, to ensure that the rerender happened.

I tried with requestionAnimationFrame, it executes before the "width: auto", resulting in nothing